compiler/parser/RdrHsSyn.hs

```diff
@@ -3432,8 +3400,12 @@ interfaceTypeRepExp = mkQualVar $ mkVarOcc "interfaceTypeRep"
 mkInterfaceDecl
   :: Located RdrName -> [Located RdrName] -> [Located InterfaceBodyDecl] -> P (OrdList (LHsDecl GhcPs))
 mkInterfaceDecl tycon requires decls = do
-    let existential :: LHsDecl GhcPs
-        existential = cL (getLoc tycon) $ TyClD noExt $ DataDecl
+    let nloc = getLoc tycon
+        mkTyCl = mkLHsSigType . rdrNameToType . L nloc . qualifyDesugar . mkClsOcc
+        derivingTys = L nloc $ map mkTyCl ["Eq"]
+        derivingClause = L nloc $ HsDerivingClause noExt Nothing derivingTys
+        existential :: LHsDecl GhcPs
+        existential = L nloc $ TyClD noExt $ DataDecl
```

Comment:
I don’ think this works. We still cannot derive the instance. The type is defined as
```
data I = I Opaque
```
What we can do now is to drop the type rep stuff and only call `mkPrimitive "primitive" "BEEqual"`.